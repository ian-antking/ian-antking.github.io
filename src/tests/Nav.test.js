import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import Nav from '../components/Nav';

describe('Nav', () => {
  it('renders the navbar', () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router history={history}>
        <Nav />
      </Router>
    );
    expect(asFragment(<Nav />)).toMatchSnapshot();
  });
});
