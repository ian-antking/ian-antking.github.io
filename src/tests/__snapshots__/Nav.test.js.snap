// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav correctly renders the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="sc-AxjAm fLNzde"
    data-testid="nav-bar"
  >
    <div
      class="sc-AxiKw gQKflZ"
    >
      <h1
        class="sc-AxirZ eMiaWZ"
        data-testid="home-link"
      >
        [IK]
      </h1>
    </div>
    <div
      class="sc-AxhCb dBIewh"
    >
      <a
        class="sc-AxhUy dPDQiT"
        data-testid="about-link"
        href="/about"
      >
        About
      </a>
      <a
        class="sc-AxhUy dPDQiT"
        data-testid="projects-link"
        href="/projects"
      >
        Projects
      </a>
      <a
        class="sc-AxhUy dPDQiT"
        data-testid="contact-link"
        href="/contact"
      >
        Contact
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
